lib/poison/decoder.ex

```diff
@@ -52,14 +52,22 @@ defmodule Poison.Decode do
   end
 
   defp do_transform_struct(value, keys, as, options) do
+    default = struct(as.__struct__)
+
     Map.from_struct(as)
     |> Enum.reduce(%{}, fn {key, as}, acc ->
-      case Map.get(value, key) do
+      new_value = case Map.get(value, key, Map.get(default, key)) do
         value when is_map(value) or is_list(value) ->
-          Map.put(acc, key, transform(value, keys, as, options))
-        value ->
-          Map.put(acc, key, value)
+          # value == as indicates we never actually recieved a value for `key`.
+          if value == as do
+            Map.get(default, key)
```

Comment:
@devinus re-reading this I have a much more performant idea in mind, as doing a Map.get(default, key) on every lookup is a bit silly. Expect a new PR in a few minutes.